core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -77,7 +77,14 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
     new File(subDir, filename)
   }
 
-  def getFile(blockId: BlockId): File = getFile(blockId.name)
+  def getFile(blockId: BlockId, shuffleGenerationId: Int = -1): File = {
+    val generationId = if (shuffleGenerationId != -1) {
+      "_" + shuffleGenerationId
+    } else {
+      ""
+    }
+    getFile(blockId.name.replace(".", generationId + "."))
```

Comment:
this looks hacky. Can we do this at the caller side when we need the `shuffleGenerationId`? Or we can create a `getFile(shuffleId: Int, shuffleGenerationId: Int, ...)` method if there are many callers.

Reply:
Copy that and thanks for the advice, I do this at the caller side and revert the changes in DiskBlockManager in 2362512.